test/terraform_aws_ecs_service_test.go

```diff
@@ -30,3 +42,131 @@ func TestTerraformAwsEcsServiceSimple(t *testing.T) {
+	taskList, err := GetTasksE(t, region, clusterName)
+	require.NoError(t, err)
+	return taskList
+}
+
+func GetTasksE(t *testing.T, region string, clusterName string) (*ecs.ListTasksOutput, error) {
+	var returnTaskList *ecs.ListTasksOutput
+	ecsClient, err := aws.NewEcsClientE(t, region)
+	if err != nil {
+		return returnTaskList, err
+	}
+
+	params := &ecs.ListTasksInput{
+		Cluster: awssdk.String(clusterName),
+	}
+
+	// Need to spin and wait to allow time for resources to get up
+	maxRetries := 3
+	retryDuration, _ := time.ParseDuration("30s")
+	_, err = retry.DoWithRetryE(t, "Get tasks", maxRetries, retryDuration,
+		func() (string, error) {
+			tasks, _ := ecsClient.ListTasks(params)
+
+			if len(tasks.TaskArns) == 0 {
+				return "Did not retrieve tasks", fmt.Errorf("We returned empty tasks %v", tasks.TaskArns)
+			}
+			returnTaskList = tasks
+			return "Retrieved tasks", nil
+		},
+	)
+	if err != nil {
+		return returnTaskList, err
+	}
+
+	return returnTaskList, nil
+}
+
+// Retrieve ENI from task ARNs and cluster
+func GetEni(t *testing.T, region string, cluster string, taskArns []*string) *string {
+	eni, err := GetEniE(t, region, cluster, taskArns)
+	require.NoError(t, err)
+	return eni
+}
+
+func GetEniE(t *testing.T, region string, cluster string, taskArns []*string) (*string, error) {
+	ecsClient, err := aws.NewEcsClientE(t, region)
+	if err != nil {
+		return nil, err
+	}
+
+	params := &ecs.DescribeTasksInput{
+		Cluster: awssdk.String(cluster),
+		Tasks:   taskArns,
+	}
+	returnedTasks, err := ecsClient.DescribeTasks(params)
+	if err != nil {
+		return nil, err
+	}
+
+	eniDetail := returnedTasks.Tasks[0].Attachments[0].Details[1].Value
+	return eniDetail, nil
+}
+
+// Retrieve Public IP from ENI
+func GetPublicIP(t *testing.T, region string, enis []string) *string {
+	publicIP, err := GetPublicIPE(t, region, enis)
+	require.NoError(t, err)
+	return publicIP
+}
+
+func GetPublicIPE(t *testing.T, region string, enis []string) (*string, error) {
+	ec2Client := aws.NewEc2Client(t, region)
+
+	params := &ec2.DescribeNetworkInterfacesInput{
+		NetworkInterfaceIds: awssdk.StringSlice(enis),
+	}
+	eniDetail, err := ec2Client.DescribeNetworkInterfaces(params)
+	if err != nil {
+		return nil, err
+	}
+
+	publicIP := eniDetail.NetworkInterfaces[0].Association.PublicIp
+	return publicIP, nil
+}
+
+func TestTerraformAwsEcsServiceContainer(t *testing.T) {
+	t.Parallel()
+
+	tempTestFolder := test_structure.CopyTerraformFolderToTemp(t, "../", "examples/container-image")
+
+	ecsServiceName := fmt.Sprintf("terratest-container-%s", strings.ToLower(random.UniqueId()))
+	awsRegion := "us-west-2"
+	vpcAzs := aws.GetAvailabilityZones(t, awsRegion)[:3]
+
+	terraformOptions := &terraform.Options{
+		// The path to where our Terraform code is located
+		TerraformDir: tempTestFolder,
+
+		// Variables to pass to our Terraform code using -var options
+		Vars: map[string]interface{}{
+			"test_name": ecsServiceName,
+			"vpc_azs":   vpcAzs,
```

Comment:
```suggestion
			"vpc_azs":   vpcAzs,
			"region":    awsRegion,
```
Removing the default value for the variable means you'll need to pass it here.